vercel/turborepo
# chore: update `basic` example to ESLint v9 and Flat Configurations.

### Description

This PR converts our `basic` example to ESLint v9 with Flat Configurations.

I may have a few gray hairs now, but I'm very excited for the community to have this.

### Testing Instructions

Use `npx create-turbo@canary -e https://github.com/vercel/turborepo/tree/shew-409f3/examples/basic basic-eslint-9` to try it out.

- Pull it down, make sure you see diagnostics where you'd expect in your editor. Including for `eslint-config-turbo`, most importantly!
- I know the diff is big, but there may be vestigials from my experimenting lying around. Please use a finely toothed comb! (I've looked myself but the more eyes, the better.)
